Confusing branches were fixed. It was harder than I expected, so solu…

Y-Pazniak opened the pull request:
…tion is very stupid, but at least it works.

Y-Pazniak committed:
Confusing branches were fixed. It was harder than I expected, so solution is very stupid, but at least it works.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/PassengerStepsChecker.java:
```diff
@@ -0,0 +1,55 @@
+
+
+    private boolean isQueueWrong(final Enum[] steps) {
+        if ((steps[0].equals(CountryOrigin.EAES) || steps[1].equals(OwnersType.PHYSICAL)) && (steps[3] != null || steps[4] != null)) {
```
```suggestion
        return (steps[0].equals(CountryOrigin.EAES) || steps[1].equals(OwnersType.PHYSICAL)) && (steps[3] != null || steps[4] != null)
```

Y-Pazniak replied:
We don't need this anymore. 
The line was too big for processing by my small brain, so I tried to split this line in simplier parts is...else

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/PassengerStepsChecker.java:
```diff
@@ -0,0 +1,55 @@
+        } else return false;
+    }
+
+    private boolean containsNullBeforeFilledNodes(final Enum[] steps) {
```
```suggestion
    private boolean isStepMissed(final Enum[] steps) {
```

Y-Pazniak replied:
Sorry for wrong naming. I try to improve myself in it

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/PassengerStepsChecker.java:
```diff
@@ -0,0 +1,55 @@
+
+import java.io.Serializable;
+
+public class PassengerStepsChecker implements Serializable {
```
Why `Serializable` is needed?

Y-Pazniak replied:
It was my mistake - I saved PassengerStepsChecker in UserProgress and UserProgress needed to be saved by serialization. 
After POJO article I realized my mistake and I hope I wont do it again.

MaximPilipeiko replied:
Please, fix it

Y-Pazniak replied:
This class has been removed. Don't need it anmyore

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -12,17 +14,23 @@ public class UserProgress implements Serializable {
     private final String chatID;
     private Step currentQuestion;
+    private Enum[] stepsData;
+    private PassengerStepsChecker passengerStepsChecker;
```
This field is redundant.  UserProgress should store only user data. Look POJO object https://naveen-metta.medium.com/unleashing-the-power-of-plain-old-java-objects-pojos-in-modern-development-19cb57638f6

Y-Pazniak replied:
The article was wonderful. The main thought is "use POJO for data-classes". That's why we don't need Enum[] and StepsChecker here

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -12,17 +14,23 @@ public class UserProgress implements Serializable {
     private VolumeOfEngine volumeOfEngine = null;
     private final String chatID;
     private Step currentQuestion;
+    private Enum[] stepsData;
```
I believe this could be configured based on above data

Y-Pazniak replied:
Yes, sorry

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -12,17 +14,23 @@ public class UserProgress implements Serializable {
     private VolumeOfEngine volumeOfEngine = null;
     private final String chatID;
     private Step currentQuestion;
+    private Enum[] stepsData;
```
If I right understand the intension `chosenAnswers` or just `answers` is more suitable.

Y-Pazniak replied:
I prefer to use Commands here. Anyway, my naming is wrong

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -31,8 +39,13 @@ public OwnersType getOwnersType() {
+        int stepID = 1;
+        cleanStepsAfterCurrent(stepID);
+        stepsData[stepID] = ownersType;
+        if (!passengerStepsChecker.isStepOk(stepsData)) {
```
It's better to make validation before all other logic.

Y-Pazniak replied:
yes, thanks

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/UserProgressManager.java:
```diff
@@ -32,63 +33,93 @@ public void createNewUserProgress(final String chatID) throws UtilsborException
+            userProgressStorage.save(userProgress);
+            message = messagesCreator.getSummaryAnswer(userProgress);
+        } catch (StepsQueueException exception) {
+            message = exception.getWrongStep();
```
Let's look at it together maybe it could be more suitable to move catch on higher level

Y-Pazniak replied:
And we have place for catching exceptions. This part is definetily wrong, thank you for your comment

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/exception/constants/UtilsborErrorCode.java:
```diff
@@ -3,7 +3,8 @@
     READING_FROM_FILE_EXCEPTION ("file_reading_error"),
-    SAVING_INTO_FILE_EXCEPTION ("file_saving_error")
+    SAVING_INTO_FILE_EXCEPTION ("file_saving_error"),
+    WRONG_QUEUE_EXCEPTION ("wrong_users_queue_steps"),
```
INVALID_OWNER_TYPE,
INVALID_CAR_AGE,

Y-Pazniak replied:
Another exceptions and theis codes have been added.

Y-Pazniak committed:
I tried to fix confusing branches by using Tree.
I am not sure about realization, but now it works better than before.

Y-Pazniak committed:
There was a mistake after testing - i've fixed it

Y-Pazniak committed:
There were some mistakes after testing - i've fixed it

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/NodeStorage.java:
```diff
@@ -0,0 +1,76 @@
+        Node eaes = new Node(start, Command.EAES);
+        Node eaesPhysical = new Node(eaes, Command.PHYSICAL_PERSON);
+        Node eaesJuridical = new Node(eaes, Command.JURIDICAL_PERSON);
+        eaes.addChildren(new ArrayList<>(Arrays.asList(eaesPhysical, eaesJuridical)));
```
```suggestion
        eaes.addChildren(Arrays.asList(eaesPhysical, eaesJuridical));
```

Y-Pazniak replied:
Lists filling has been improved

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/NodeStorage.java:
```diff
@@ -0,0 +1,76 @@
+        eaesPhysical.addChildren(new ArrayList<>(List.of(eaesPhysicalEnd)));
+        Node eaesJuridicalEnd = new Node(eaesJuridical, Command.AGE);
+        eaesJuridical.addChildren(new ArrayList<>(List.of(eaesJuridicalEnd)));
+        Collections.addAll(nodes, eaes, eaesPhysical, eaesJuridical, eaesPhysicalEnd, eaesJuridicalEnd);
```
Don't understand why it's needed

Y-Pazniak replied:
I guess, we discussed about it oraly. In short, I need this code to fill my tree. If we need to discuss it in details, we can do it. Also, if we have more beautifil way to fill Tree - I will be happy to learn it. Current method depends on programmers attention and it is difficult to fill the tree this way and extremly difficult to find a mistake there.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/NodeStorage.java:
```diff
@@ -0,0 +1,76 @@
+        Node other = new Node(start, Command.OTHER_COUNTRIES);
+        start.addChildren(new ArrayList<>(Arrays.asList(eaes, other)));
+        Node otherPhysical = new Node(other, Command.PHYSICAL_PERSON);
+        Node otherPhysicalEnd = new Node(otherPhysical, Command.AGE);
```
Naming is strange

Y-Pazniak replied:
In my mind, age == end, it is the last node. But I must not be so selfish and I must think about another programmers, so I renamed end to age. 

In other cases naming looks for me OK - I tried to name nodes by their way 
Node other - it means it is the node from other countries. 
Node otherPhysical - it means, this node has parent "other" and contains key "physical".
Node otherPhysicalEnd - this is node from other countries branch, whith physical person parend and "age" key inside.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/NodeStorage.java:
```diff
@@ -0,0 +1,76 @@
+        Collections.addAll(nodes, other, otherPhysical, otherPhysicalEnd);
+
+        //other juridical electric nodes queue
+        Node otherJuridical = new Node(other, Command.JURIDICAL_PERSON + "_other");
```
Don't understand this prefix `_other`

Y-Pazniak replied:
It was my huge mistake, sory. Fixed now.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/constants/Command.java:
```diff
@@ -27,4 +27,8 @@ public class Command {
     public static final String VOLUME_MORE_3500_CM = "/more_3500";
+
+    //command not for user - just for path checking
+    public static final String AGE = "age";
```
Please, explain the idea

Y-Pazniak replied:
We discussed about it orally. In short, I tried to shorter my map using additional commands, but we during the discussion we found this path as mistake, so in last PR i removed this commads.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/UserProgressManager.java:
```diff
@@ -2,27 +2,38 @@
     private final IUserProgressStorage userProgressStorage;
+    private final NodeStorage nodeStorage;
+
+    private final Tree tree;
```
I think, to have both nodeStorage and tree is redundant

Y-Pazniak replied:
We decided to remove both of it and replace it by UserProgressValidator

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/UserProgressManager.java:
```diff
@@ -32,51 +43,97 @@ public void createNewUserProgress(final String chatID) throws UtilsborException
         UserProgress userProgress;
         userProgress = userProgressStorage.get(chatID);
+        userProgress.setNode(nodeStorage.updateUserNode(command));
+        System.out.println(userProgress.getCurrentNode().getKey());
```
avoid println in production code

Y-Pazniak replied:
Removed here. 
Is it allowed to use println in try-catch or it is strict rule not to use println anywhere?

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -12,17 +19,40 @@ public class UserProgress implements Serializable {
     private final String chatID;
     private Step currentQuestion;
+    private Node node;
+    private final List<Node> nodes;
```
You could store just commands or answers

Y-Pazniak replied:
Fixed it. We implemented a way to restore user's path and his path by UserProgress fields, not even commands.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -12,17 +19,40 @@ public class UserProgress implements Serializable {
     private final String chatID;
     private Step currentQuestion;
+    private Node node;
+    private final List<Node> nodes;
```
```suggestion
    private final List<String> path;
```

Y-Pazniak replied:
Dont' need any list here anymore

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/Tree.java:
```diff
@@ -0,0 +1,36 @@
+        return TreeHolder.TREE_HOLDER;
+    }
+
+    public boolean validateNode(final Node node, final String command, final List<Node> nodes) {
```
```suggestion
    public boolean validateNode(final List<String> path, final String newCommand) {
```

Y-Pazniak replied:
Don't need any lists here. Now it looks
```
public boolean validateCommand(final String requestingCommand, final UserProgress userProgress) {
        return isRequestingCommandAcceptable(requestingCommand, getNode(userProgress));
    }
```

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/Tree.java:
```diff
@@ -0,0 +1,36 @@
+
+public class Tree {
+    private final List<Node> nodes;
+
```
```suggestion
private final Node start;
```

Y-Pazniak replied:
We don't need any fields here now

Y-Pazniak committed:
Tree has been rewritten

Y-Pazniak committed:
Tree has been rewritten

Y-Pazniak committed:
Tree has been rewritten

Y-Pazniak committed:
Tree has been rewritten

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -21,8 +28,56 @@ public CountryOrigin getCountryOrigin() {
+
+        if (getCountryOrigin() != null) {
+            if (getCountryOrigin().equals(CountryOrigin.EAES)) {
+                userPath[0] = Command.EAES;
```
What about save not command, but answer. Lets discuss it on call

Y-Pazniak replied:
Yes, we have improved this code amazingly after call. Thank you!

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/UserProgress.java:
```diff
@@ -21,8 +28,56 @@ public CountryOrigin getCountryOrigin() {
+    public String[] getUserPath() {
+        String[] userPath = new String[5];
+
+        if (getCountryOrigin() != null) {
```
It's possible to use Map instead a lot of if-else

Y-Pazniak replied:
Yes, it was a great idea. Implemented.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/TreeCommand.java:
```diff
@@ -0,0 +1,85 @@
+import java.util.Objects;
+
+public class TreeCommand {
+    private final List<Node> nodes;
```
Why do you need list of nodes?

Y-Pazniak replied:
Don't need it anymore. Removed. Completely.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/TreeCommand.java:
```diff
@@ -0,0 +1,85 @@
+        Node currentNode = nodes.getFirst();
+        String[] userPath = userProgress.getUserPath();
+        for (String userPassedStep : userPath) {
+            for (Node node : nodes) { //to recreate path we need to check every node
```
For me it's incorrect. It's not needed to check every node, only children.

Y-Pazniak replied:
Yes, you are right. For unknown reasons it is difficult to work with nodes for me, so my solutions arent the best. I've improved code by using search only between kids.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/TreeCommand.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    //hello! even I can hardly understand what is going on, so there are comments
+    private Node currentNodeCreation(final UserProgress userProgress) { //method for creating current user's step node
```
```suggestion
    private Node findCurrentNode(final UserProgress userProgress) { //method for creating current user's step node
```

Y-Pazniak replied:
What do you think about getNode()? 
I like **getNode(final UserProgress userProgress)**;

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/TreeCommand.java:
```diff
@@ -0,0 +1,85 @@
+
+    private boolean isRequestingCommandInCurrentNodeKids(final String requestingCommand, final Node currentNode) { //method to check is it permissible to get next node
+        boolean requestingCommandOk = false;
+        for (Node kidNode : currentNode.getChildren()) { //taking node's kids
```
what about to use method `include` instead go throw all nodes

MaximPilipeiko replied:
Set<String> childrenKeys = currentNode.getChildren().map((c) -> c.key).toSet()
Set.contains(requestingCommand)

Y-Pazniak replied:
Sorry, I have NOT implemented it yet. It seems clear for me, but I need time to come to awareness and accept new solution.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/UserProgressManager.java:
```diff
@@ -2,82 +2,129 @@
         userProgress = userProgressStorage.get(chatID);
-        switch (command) {
+        treeCommand.validateCommand(requestingCommand, userProgress);
+        userProgress.setCommand(requestingCommand);
```
Do we really need it?

Y-Pazniak replied:
We don't need this lines. I've removed it.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/TreeCommand.java:
```diff
@@ -0,0 +1,85 @@
+import java.util.List;
+import java.util.Objects;
+
+public class TreeCommand {
```
```suggestion
public class CommandTree {
```

Y-Pazniak replied:
During the call we agreed to leave it as is, because we have another Tree classes and all of them finish by ...Tree.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/TreeCommand.java:
```diff
@@ -0,0 +1,85 @@
+        return currentNode;
+    }
+
+    private boolean isRequestingCommandInCurrentNodeKids(final String requestingCommand, final Node currentNode) { //method to check is it permissible to get next node
```
```suggestion
    private boolean existsCommandInNodeChildren(final String command, final Node node) { //method to check is it permissible to get next node
```

Y-Pazniak replied:
Methods were renamed

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/TreeCommand.java:
```diff
@@ -0,0 +1,85 @@
+    private boolean isRequestingCommandInCurrentNodeParents(String requestingCommand, Node node) {
+        Node parent = node.getParent();
+        while (!Objects.isNull(parent)) {
+            if (requestingCommand.equals(parent.getKey())) {
```
boolean isRequestingCommandInParent = requestingCommand.equals(parent.getKey())
boolean isRequestingCommandInCurrentNodeKids = isRequestingCommandCurrentInNodeKids(requestingCommand, parent)

if (isRequestingCommandInParent || isRequestingCommandCurrentInNodeKids)

Y-Pazniak replied:
Fixed. Now it looks better. For me, it is almost perfect.

Y-Pazniak committed:
Solution with maps has been added.
Common interface has been added.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/StepsCheckerContainer.java:
```diff
@@ -0,0 +1,39 @@
+import java.util.List;
+
+public class StepsCheckerContainer {
+    private Enum[] eaesArray = {CountryOrigin.EAES, OwnersType.PHYSICAL};
```
```suggestion
    private StepsIndicator[] eaesArray = {CountryOrigin.EAES, OwnersType.PHYSICAL};
```

Y-Pazniak replied:
Fixed through all the code

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/HelperTree.java:
```diff
@@ -0,0 +1,23 @@
+            new AbstractMap.SimpleEntry<>(OwnersType.PHYSICAL, Command.PHYSICAL_PERSON),
+            new AbstractMap.SimpleEntry<>(OwnersType.JURIDICAL, Command.JURIDICAL_PERSON),
+
+            new AbstractMap.SimpleEntry<>(CarAge.ANY_AGE, Command.AGE),
```
I think add all ages

Y-Pazniak replied:
All ages and volumes has been added

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/HelperTree.java:
```diff
@@ -0,0 +1,23 @@
+
+public class HelperTree {
+    public static final Map<StepsIndicator, String> fieldsToCommands = Map.ofEntries(
+            new AbstractMap.SimpleEntry<>(CountryOrigin.EAES, Command.EAES),
```
A more elegant way to create Entry must exist

Y-Pazniak replied:
Have been replaced by put(...) and static{} block

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/CommandTree.java:
```diff
@@ -0,0 +1,86 @@
+import java.util.Objects;
+
+public class CommandTree {
+    private final List<Node> nodes;
```
Why list?

Y-Pazniak replied:
List was removed, don't need it.

Y-Pazniak committed:
Search for a current node - do not search throughout the entire stack, but only through children. We look at the list of commands, go along the sheet, recreate the nodes until we hit zero < < < DONE
Make map initialization shorter and better. The best way is to use static {put(); put()...}  < < < DONE
To fix adding commands into map - we can add there ALL commands, don't need to add additional Enum objects just for map.  < < < DONE
Search and destroy all Enum arrays - change it to StepsIndicator interface.  < < < DONE
userProgress[]  - to find a way to remove ternar operations. It is possible to make a videos and discuss it with Maksim.  < < < DONE
Try to add null into map - it is possible, it can help us to fix problem with userProgress[] ternar
Create a layer between UserProgressManager and Tree. Name it UserProgressValidator. Design it not as a singleton, but as a static one. Make a Tree field inside the UserProgressValidator; it should also be static. < < < DONE
Fix bugs in the Tree pull request. < < < DONE (one left)
Github - find all old comments before map and fix it. All of them.  < < < DONE (one left)

Y-Pazniak committed:
stream().map() instead foreach for kids key searching added

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/Node.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Node implements Serializable {
```
```suggestion
public class Node {
```

Y-Pazniak replied:
Fixed

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/CommandTree.java:
```diff
@@ -0,0 +1,75 @@
+import java.util.stream.Collectors;
+
+public class CommandTree {
+
```
I believe head of tree should be placed here not in `NodeStorage`

Y-Pazniak replied:
NodeStorage has been removed

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/NodeStorage.java:
```diff
@@ -0,0 +1,68 @@
+        return nodes;
+    }
+
+    private void fillNodes() {
```
Next step, fill nodes from files

Y-Pazniak replied:
Noticed it in trello

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/NodeStorage.java:
```diff
@@ -0,0 +1,68 @@
+    private void fillNodes() {
+        //start node
+        Node start = new Node(null, Command.START);
+        nodes.add(start);
```
`nodes` is redundant I think

Y-Pazniak replied:
List has been removed

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/exception/OwnerTypeException.java:
```diff
@@ -0,0 +1,24 @@
+import by.custom.utilcalculator.exception.constants.UtilsborErrorCode;
+import by.custom.utilcalculator.exception.constants.UtilsborErrorDescription;
+
+public class OwnerTypeException extends UtilsborException {
```
Are you sure that you need exception for every step?

Y-Pazniak replied:
Don't need it, replaced by "InvalidOrderAnswerException"

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/CommandTree.java:
```diff
@@ -0,0 +1,75 @@
+public class CommandTree {
+
+    private CommandTree() {
+    }
```
```suggestion
        root = HelperTree.buildTree()
    }
```

Y-Pazniak replied:
Root moved to the CommandTree class, creates by HelperTree.buildTree()

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/CommandTree.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    private boolean existCommandInNodeChildren(final String requestingCommand, final Node currentNode) {
+        Set<String> childrenKeys = currentNode.getChildren().stream().map(Node::getKey).collect(Collectors.toSet()); //I literally have no idea what is it and how it works O_O
```
```suggestion
        Set<String> childrenKeys = currentNode.getChildren().stream().map((node) -> node.getKey()).collect(Collectors.toSet()); //I literally have no idea what is it and how it works O_O
```

Y-Pazniak committed:
Remove NodeStorage class and list. Replace building tree into HelperTree class - DONE
Rename filling tree method into buildTree and place all the logic there - DONE
Cut exceptions - leave only one InvalidOrderException - DONE

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/HelperTree.java:
```diff
@@ -0,0 +1,67 @@
+
+public class HelperTree {
+    public static final Map<StepsIndicator, String> fieldsToCommands = new HashMap<>(14);
+    public static final Node treeRoot = new Node(null, Command.START);
```
Please, move it to CommandTree

Y-Pazniak replied:
Done

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/exception/InvalidOrderAnswerException.java:
```diff
@@ -0,0 +1,24 @@
+import by.custom.utilcalculator.exception.constants.UtilsborErrorCode;
+import by.custom.utilcalculator.exception.constants.UtilsborErrorDescription;
+
+public class InvalidOrderAnswerException extends UtilsborException {
```
I think `InvalidOrderCommandException` is better. It's better to avoid extending business language if it's possible.

Y-Pazniak replied:
Yes, it's better. Done

Y-Pazniak committed:
InvalidOrderCommandException  > renamed.
Map<StepsIndicator, String> fieldsToCommands, Node treeRoot removed to the CommandTree class

Y-Pazniak committed:
System variables have been added.
Branches have been corrected according to the new law.
Prices have been corrected accrording to new law

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/CommandTree.java:
```diff
@@ -0,0 +1,83 @@
+        return isRequestingCommandAcceptable(requestingCommand, getNode(userProgress));
+    }
+
+    public boolean isRequestingCommandAcceptable(String requestingCommand, Node node) {
```
And do it every where. I'm pretty sure that it's possible to configure IDE to verify it
```suggestion
    public boolean isRequestingCommandAcceptable(final String requestingCommand, final Node node) {
```

Y-Pazniak replied:
I haven't known about cleanup settings yet. Thanks a lot! This settings are awesome! 
![Code Cleanup Inspections_240415214758](https://github.com/Y-Pazniak/custom_calculator_chatbot/assets/73634869/cfb89694-3597-4984-8a7b-b7ba3a7f3e64)

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/CommandTree.java:
```diff
@@ -0,0 +1,83 @@
+        return isRequestingCommandInParent || isRequestingCommandInKids || isRequestingCommandEqualsNode;
+    }
+
+    private boolean isRequestingCommandEqualsCurrentNode(String requestingCommand, Node node) {
```
```suggestion
    private boolean doesRequestingCommandEqualCurrentNode(String requestingCommand, Node node) {
```

Y-Pazniak replied:
Fixed

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/domain/tree/HelperTree.java:
```diff
@@ -0,0 +1,69 @@
+import java.util.*;
+
+public class HelperTree {
+    public static Map<StepsIndicator, String> fillMap() {
```
Which map to fill? It's too abstract. I suggest `fillFieldsToCommandsMap`.

Y-Pazniak replied:
Renamed, thanks

Y-Pazniak committed:
Methods params were corrected to final
Methods names were fixed (grammar mistakes, too abstract naming)
